src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolver.cs

```diff
@@ -152,21 +152,24 @@ internal override string GetRequestModuleName(DkmRuntimeFunctionResolutionReques
             return request.ModuleName;
         }
 
-        internal override void OnFunctionResolved(
-            DkmClrModuleInstance module,
-            DkmRuntimeFunctionResolutionRequest request,
-            int token,
-            int version,
-            int ilOffset)
-        {
-            var address = DkmClrInstructionAddress.Create(
-                module.RuntimeInstance,
-                module,
-                new DkmClrMethodId(Token: token, Version: (uint)version),
-                NativeOffset: uint.MaxValue,
-                ILOffset: (uint)ilOffset,
-                CPUInstruction: null);
-            request.OnFunctionResolved(address);
+        private static OnFunctionResolvedDelegate<DkmClrModuleInstance, DkmRuntimeFunctionResolutionRequest> OnFunctionResolved(DkmWorkList workList)
+        {
+            return (DkmClrModuleInstance module,
+                        DkmRuntimeFunctionResolutionRequest request,
+                        int token,
+                        int version,
+                        int ilOffset) =>
+            {
+                var address = DkmClrInstructionAddress.Create(
+                    module.RuntimeInstance,
+                    module,
+                    new DkmClrMethodId(Token: token, Version: (uint)version),
+                    NativeOffset: 0,
+                    ILOffset: (uint)ilOffset,
+                    CPUInstruction: null);
+                // Use async overload of OnFunctionResolved to avoid deadlock.
+                request.OnFunctionResolved(workList, address, result => { });
```

Comment:
@plnelson out of curiosity, is there ever a scenario where a meaningful completion routine would be passed to ```OnFunctionResolved```?  (is it just there to satisfy the template/pattern, or is there a meaningful scenario for it?)